python/foglamp/plugins/south/sensehat/sensehat.py

```diff
@@ -0,0 +1,262 @@
+    """
+    sense = SenseHat()
+    # The IMU (inertial measurement unit) sensor is a combination of three sensors, each with an x, y and z axis
+    # Enables and disables the magnetometer, accelerometer, gyroscope
+    sense.set_imu_config(_str_to_bool(handle['magnetometerSensor']['value']),
+                         _str_to_bool(handle['gyroscopeSensor']['value']),
+                         _str_to_bool(handle['accelerometerSensor']['value']))
+    time_stamp = utils.local_timestamp()
+    data = list()
+    asset_prefix = handle['assetNamePrefix']['value']
+    try:
+        if handle['pressureSensor']['value'] == 'true':
+            pressure = sense.get_pressure()
+            data.append({
+                'asset': '{}{}'.format(asset_prefix, handle['pressureSensorName']['value']),
+                'timestamp': time_stamp,
+                'key': str(uuid.uuid4()),
+                'readings': {"pressure": pressure}
+            })
+        if handle['temperatureSensor']['value'] == 'true':
+            temperature = sense.get_temperature()
+            data.append({
+                'asset': '{}{}'.format(asset_prefix, handle['temperatureSensorName']['value']),
+                'timestamp': time_stamp,
+                'key': str(uuid.uuid4()),
+                'readings': {"temperature": temperature}
+            })
+        if handle['humiditySensor']['value'] == 'true':
+            humidity = sense.get_humidity()
+            data.append({
+                'asset': '{}{}'.format(asset_prefix, handle['humiditySensorName']['value']),
+                'timestamp': time_stamp,
+                'key': str(uuid.uuid4()),
+                'readings': {"humidity": humidity}
+            })
+        if handle['magnetometerSensor']['value'] == 'true':
+            magnetometer = sense.get_compass_raw()
+            data.append({
+                'asset': '{}{}'.format(asset_prefix, handle['magnetometerSensorName']['value']),
+                'timestamp': time_stamp,
+                'key': str(uuid.uuid4()),
+                'readings': magnetometer
+            })
+        if handle['gyroscopeSensor']['value'] == 'true':
+            gyroscope = sense.get_gyroscope_raw()
+            data.append({
+                'asset': '{}{}'.format(asset_prefix, handle['gyroscopeSensorName']['value']),
+                'timestamp': time_stamp,
+                'key': str(uuid.uuid4()),
+                'readings': gyroscope
+            })
+        if handle['accelerometerSensor']['value'] == 'true':
+            accelerometer = sense.get_accelerometer_raw()
+            data.append({
+                'asset': '{}{}'.format(asset_prefix, handle['accelerometerSensorName']['value']),
+                'timestamp': time_stamp,
+                'key': str(uuid.uuid4()),
+                'readings': accelerometer
+            })
+
+    except (Exception, RuntimeError) as ex:
+        _LOGGER.exception("Sense HAT exception: {}".format(str(ex)))
```

Comment:
I suggest to handle `RuntimeError as e` in other block, you don't need str(ex). always use %s for logger